### PR TITLE
option to turn on/off adm borders

### PR DESCRIPTION
**Pull request by camilabarca**

### CODE REVIEW

File: projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts [hunk 1]
```diff
@@ -637,6 +637,7 @@ export class RehabPrioComponent implements OnInit {
     // ADM Analysis / Staleness
     const admanView = props.mode === 'adman' ? props.adman_view : (props.mode === 'staleness' ? 'staleness' : '');
     const admanLevel= props.adman_level || 'best';
+    this.rpState.show_adm_borders = props.mode === 'adman' ? false : (props.mode === 'staleness' ? false : (this.rpState.show_adm_borders));
```
Comment (akariv) [inline projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts]: don't overwrite the state here - use a local variable.

File: projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts [hunk 2]
```diff
@@ -737,6 +738,18 @@ export class RehabPrioComponent implements OnInit {
         this.rpState.map.setLayoutProperty(layer.id, 'visibility', this.rpState.show_landcover ? 'visible' : 'none');
       }
     });
+
+    if (this.rpState.show_adm_borders){
+
+      this.rpState.map.setFilter('adm-analysis-borders', null);
+      this.rpState.map.setPaintProperty('adm-analysis-borders', 'line-color','#5D3FD3');
+      this.rpState.map.setPaintProperty('adm-analysis-borders', 'line-opacity', 0.2);
+      this.rpState.map.setLayoutProperty('adm-analysis-borders', 'visibility', 'visible');
```
Comment (akariv) [inline projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts:765]: you also need to turn them off if show_adm_borders is false and we're not in adman/staleness modes.

File: projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts [hunk 3]
```diff
@@ -739,14 +739,18 @@ export class RehabPrioComponent implements OnInit {
       }
     });
 
-    if (this.rpState.show_adm_borders){
-
+    if (admBorders){
       this.rpState.map.setFilter('adm-analysis-borders', null);
```
Comment (akariv) [inline projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts]: we don't need to clear the filter here, right?

File: projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts [hunk 4]
```diff
@@ -739,14 +739,18 @@ export class RehabPrioComponent implements OnInit {
...
       this.rpState.map.setPaintProperty('adm-analysis-borders', 'line-color','#5D3FD3');
       this.rpState.map.setPaintProperty('adm-analysis-borders', 'line-opacity', 0.2);
       this.rpState.map.setLayoutProperty('adm-analysis-borders', 'visibility', 'visible');
+    }
 
-
+    if (!this.rpState.show_adm_borders){
+      this.rpState.map.setFilter('adm-analysis-borders', null);
```
Comment (akariv) [inline projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts]: why do you clear the filter here?

File: projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts [hunk 5]
```diff
@@ -739,14 +739,18 @@ export class RehabPrioComponent implements OnInit {
...
+      this.rpState.map.setPaintProperty('adm-analysis-borders', 'line-color','#000000');
```
Comment (akariv) [inline projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts]: setting the line color and opacity is not needed if the entire layer is not visible...

File: projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts [hunk 6]
```diff
@@ -739,14 +739,18 @@ export class RehabPrioComponent implements OnInit {
...
-
+    if (admBorders){
       this.rpState.map.setFilter('adm-analysis-borders', null);
       this.rpState.map.setPaintProperty('adm-analysis-borders', 'line-color','#5D3FD3');
       this.rpState.map.setPaintProperty('adm-analysis-borders', 'line-opacity', 0.2);
       this.rpState.map.setLayoutProperty('adm-analysis-borders', 'visibility', 'visible');
+    }
 
-
+    if (!this.rpState.show_adm_borders){
```
Comment (akariv) [inline projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts]: why not use `else` here? or at least `if (!admBorders) {`?

**Review by akariv — CHANGES_REQUESTED** [review]

File: projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts (hunk 3 above)
Comment (camilabarca) [reply projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts]: If I don't clear the filter hear the filter from adm analysis stays

File: projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts [hunk 7]
```diff
@@ -673,6 +673,7 @@ export class RehabPrioComponent implements OnInit {
       this.rpState.map.setPaintProperty('adm-analysis', 'fill-color', interpolate);
       this.rpState.map.setPaintProperty('adm-analysis-borders', 'line-color', ['to-color', this.colorRange[4]]);
       this.rpState.map.setPaintProperty('adm-analysis-borders', 'line-opacity', 1);
+      this.rpState.map.setLayoutProperty('adm-analysis-borders', 'visibility', 'visible');
```
Comment (akariv) [inline projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts]: what's the point? It gets re-set anyway in line 692 right after this...

**Review by akariv — CHANGES_REQUESTED** [review]

File: projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts (hunk 3 above)
Comment (akariv) [reply projects/wpdx/src/app/rehab-prio/rehab-prio.component.ts]: yes, but note that the filter is made up of adm_level filtering as well as the filter by region.
while we indeed want to show all adm levels, we still want to keep filtering by region...

**Review by akariv — APPROVED** [review]

**Pull request merged**